contracts/CAT_FACTORY.sol

```diff
@@ -0,0 +1,67 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/Counters.sol";
+import "./ContributionAccountingToken.sol"; // Make sure this import points to your CAT contract file
+
+contract CATFactory is Ownable {
+    using Counters for Counters.Counter;
+
+    Counters.Counter private _tokenIds;
+
+    // Mapping from owner address to token addresses
+    mapping(address => address[]) private _ownedTokens;
+
```

Comment:
Let's add another mapping called `_mintableTokens`

Reply:
The first mapping returns the tokens for which the user has `admin` role. The second mapping returns the the tokens for which the user has `minter` role.

Reply:
When admin gives the minter role to a new address in the CAT, the CAT contract should call the factory contract to update this mapping.

Reply:
The "My CATs" page should show all CATs for which a user has either `admin` or `minter` role.